pkg/controller/certificates/signer/signer.go

```diff
@@ -44,56 +46,37 @@ func NewCSRSigningController(
 	if err != nil {
 		return nil, err
 	}
+
 	return certificates.NewCertificateController(
 		"csrsigning",
 		client,
 		csrInformer,
 		signer.handle,
-	), nil
+	).WithDynamicCertReloader(signer.caLoader), nil
 }
 
 type signer struct {
-	ca      *authority.CertificateAuthority
+	caValue  atomic.Value
+	caLoader *dynamiccertificates.DynamicFileServingContent
```

Comment:
Can you create a separate struct (e.g. "caProvider") to encompass this functionality?